### PR TITLE
migration example for javascript

### PR DESCRIPTION
**Pull request by aricart**
Note that I am not sure why we have node/deno in this particular case the code is is exactly the same regardless of the runtime - as it is the same underlying NBC library. Possibly we simply rename "deno" as "javascript" and omit the node

@Jarema - I was looking at the Go example and it is very good, but I am wondering if we need less information about the legacy stuff - effectively if we document legacy when trying to document the new one, we have a distraction and seems harder than it should be. WDYT?

### CODE REVIEW

**Comment by Jarema** [comment]
@aricart hm, that's a good point. 
It is very detailed, and @piotrpio did great job there, but I agree that we should focus rather on explaining the new API.
I think we should make it a little more lightweight. Just explain the minimum in legacy, relevant for the new API differences.

**Comment by aricart** [comment]
I changed mine to do less of that, but I think there's room for even less.

**Comment by piotrpio** [comment]
@aricart that's a good point. I will modify my example to have less noise about the legacy stuff.

File: examples/jetstream/api-migration/deno/main.ts [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+// on a consumer:
+const js = nc.jetstream();
+// Here we add 20 messages to a stream
+const proms = Array.from({ length: 20 })
+  .map((_v, idx) => {
+    return js.publish(`events.${idx}`);
+  });
+await Promise.all(proms);
+
+// Processing messages using Legacy JetStream is fairly cumbersome
```
Comment (piotrpio) [inline examples/jetstream/api-migration/deno/main.ts:35]: Unfortunately those side notes will not be rendered correctly in NBE, so they should reside above some code for which they make sense. Right now if you build the example and run `nbe serve`, the way this is rendered looks like this ("Processing messages..." sits right next to the `subscribe` example code):


image

File: examples/jetstream/api-migration/deno/main.ts [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+const pushSub = await js.subscribe("events.>", opts);
+for await (const m of pushSub) {
+  console.log(`legacy push subscriber received ${m.subject}`);
+  m.ack();
+  // we can check if the stream has more messages
+  // by checking the number of pending messages, and break
+  // if we are done.
+  if (m.info.pending === 0) {
+    // breaking from the iterator will stop the subscription
```
Comment (piotrpio) [inline examples/jetstream/api-migration/deno/main.ts:60]: This does not render nicely:

image

File: examples/jetstream/api-migration/deno/main.ts [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+});
+
+// The different ways of getting messages from the consumers are there
+// to help you align the buffering requirements of your application with
+// what the client is doing.
+
+// The legacy API provided a way of retrieving a single message:
+const m = await js.pull("EVENTS", "my-durable")
+  .catch((err) => {
+    // possibly no messages found
```
Comment (piotrpio) [inline examples/jetstream/api-migration/deno/main.ts:159]: Thioe comments in block do not look nicely in rendered example (same applies to other inline comments in this example):

image

File: examples/jetstream/api-migration/deno/main.ts [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+
+// To prevent those sort of issues, the legacy API also provided a
+// `pullSubscribe()`, which effectively avoided the issues of the push
+// (where the client can be overwhelmed with messages). However, this
+// API is hard to use:
+opts = consumerOpts()
+  .ackExplicit()
+  .manualAck();
+
+// pullSubscribe() would create a subscription to process messages
```
Comment (piotrpio) [inline examples/jetstream/api-migration/deno/main.ts:84]:
```suggestion
// `pullSubscribe()` would create a subscription to process messages
```

**Review by piotrpio — COMMENTED** [review]
Looks very nice, other than some nbe specific rendering issues

**Comment by bruth** [comment]
> Note that I am not sure why we have node/deno in this particular case the code is is exactly the same regardless of the runtime - as it is the same underlying NBC library. Possibly we simply rename "deno" as "javascript" and omit the node

Also re: ☝️ I agree they should be consolidated. I suppose the client could be "JavaScript" and we could standardize on either runtime and using the correct import statements. For the other runtime, it could be simply commented out "If using Node, use this import instead.." for example.

**Comment by aricart** [comment]
@bruth I will need your help with some of the formatting, nbe picks the first comment following a code block to line up with the next block, rather than defer to its immediate next. But perhaps there's a trick

**Pull request merged**